eth_utils/address.py

```diff
@@ -107,16 +108,18 @@ def to_checksum_address(value: AnyStr) -> ChecksumAddress:
     Makes a checksum address given a supported format.
     """
     norm_address = to_normalized_address(value)
-    address_hash = encode_hex(keccak(text=remove_0x_prefix(norm_address)))
+    address_hash = encode_hex(keccak(text=remove_0x_prefix(HexStr(norm_address))))
 
     checksum_address = add_0x_prefix(
-        "".join(
-            (
-                norm_address[i].upper()
-                if int(address_hash[i], 16) > 7
-                else norm_address[i]
+        HexStr(
+            "".join(
+                (
+                    norm_address[i].upper()
+                    if int(address_hash[i], 16) > 7
+                    else norm_address[i]
+                )
+                for i in range(2, 42)
```

Comment:
While you're in here, can you update this to do:

```python
(
    digit.upper()
    if digit in DIGITS_8_THROUGH_15
    else digit
    for digit in norm_address[2:42]
```

I think it has a slight impovement to both readability and performance.

Reply:
I'm not so sure this works out to the same thing. Since we should use digits from  the `address_hash` on line 118 to determine whether or not to `.upper()` the digit from the `norm_address`. I think we need to index into the original `norm_address` regardless.

Reply:
Ignore me,  you're correct, I just glanced and thought all three indexes were from the same structure.